### ISSUE TITLE
On Slack 4, it get stuck starting, it just shows the Slack Quote, and then nothing

### ISSUE THREAD

**Issue by patroza** [LanikSJ/slack-dark-mode#72]
**Describe the bug**
Slack no longer starts after installing dark mode.

**To Reproduce**
1. checkout slack-4.0 branch
2. install
3. restart Slack
4. only see Slack Quote at startup, then stuck

**Expected behavior**
Dark theme loads and slack loads

**Desktop (please complete the following information):**

-   OS: macos 10.14.5

**Comment by patroza** [comment]
It looks like there are 2 issues in the ssb js file:
1. There is a `.js.map` comment on the end, and you're appending the script to that, need first a line break
2. The sed command literally replaces the path with `"$THEME_PATH"` instead of with the value of that variable.

**Comment by sudochownrwhoami** [comment]
Just wrote the solution and tried pushing to a new branch, but I guess I can't create a branch and create a PR? 
@LanikSJ I assume you can help out here? It wasn't a quick fix, and I did spend a bit of time working on this, so I'd appreciate it if I could create the PR. It does need testing on linux though.
This is one of my first open-source contributions (of many to come), so perhaps I'm not going about this the right way? lmk, thanks!
Anyway, here's the newer `slack-dark-mode.sh` I ended up with:
```sh
#!/usr/bin/env bash

# Homebaked Slack Dark Mode. After executing this script, hit refresh (⌘ + R) or restart Slack for changes to take effect.
# Adopted from https://gist.github.com/a7madgamal/c2ce04dde8520f426005e5ed28da8608

MAC_OS_SLACK_RESOURCES_DIR="/Applications/Slack.app/Contents/Resources"
LINUX_SLACK_RESOURCES_DIR="/usr/lib/slack/resources"
DARK_THEME_PATH_PLACEHOLDER="SLACK_DARK_THEME_PATH"
DARK_THEME_FILENAME="dark-theme.css"
UPDATE_THEME_ONLY="false"

file_exists(){
	# check if $1 exists and also protect against an empty string
    if [[ -f $1 ]] && [[ -n $1 ]] ; then
        return 0
    else
        return 1
    fi
}

is_directory(){
	# check if $1 is a directory and also protect against an empty string
    if [[ -d $1 ]] && [[ -n $1 ]] ; then
        return 0
    else
        return 1
    fi
}

# carriage return equivalent then echo arg
cr_echo(){
	echo && echo $1
}

if is_directory $MAC_OS_SLACK_RESOURCES_DIR ; then SLACK_RESOURCES_DIR=$OSX_SLACK_RESOURCES_DIR; fi
if is_directory $LINUX_SLACK_RESOURCES_DIR; then SLACK_RESOURCES_DIR=$LINUX_SLACK_RESOURCES_DIR; fi
if [[ "$1" == "-u" ]]; then UPDATE_THEME_ONLY="true"; fi

SLACK_EVENT_LISTENER="event-listener.js"
SLACK_DIST_DIR="$SLACK_RESOURCES_DIR/app.asar.unpacked/dist"
SLACK_INTEROP_FILENAME="ssb-interop.bundle.js"
SLACK_INTEROP_FILEPATH="$SLACK_DIST_DIR/$SLACK_INTEROP_FILENAME"
THEME_FILEPATH="$SLACK_RESOURCES_DIR/$DARK_THEME_FILENAME"

if [[ "$UPDATE_THEME_ONLY" == "true" ]]; then
  cr_echo "Updating Dark Theme Code to Slack... "
else
  cr_echo "Adding Dark Theme Code to Slack... "
fi

echo "This script requires sudo privileges."
echo "You'll need to provide your password."

# Copy CSS to Slack Folder
sudo cp -af dark-theme.css "$THEME_FILEPATH"

# Unpack Asar Archive for Slack
sudo npx asar extract $SLACK_RESOURCES_DIR/app.asar $SLACK_RESOURCES_DIR/app.asar.unpacked

# Add JS Code to Slack
if [[ "$UPDATE_ONLY" == "true" ]]; then
  cr_echo "Updating Dark Theme Code to Slack... "
else
  cr_echo "Adding Dark Theme Code to Slack... "
  # We're appending a newline below to avoid appending commented lines, etc.
  # In particular, without this (or without adding an empty first line to SLACK_EVENT_LISTENER), Slack freezes on macos
  TEMP_NEWLINE_FILE=/tmp/newline.txt
  sudo echo "" > $TEMP_NEWLINE_FILE
  sudo tee -a "$SLACK_INTEROP_FILEPATH" < $TEMP_NEWLINE_FILE
  sudo tee -a "$SLACK_INTEROP_FILEPATH" < $SLACK_EVENT_LISTENER
fi

# sed is not pleased when a string substitution has unescaped backslashes
THEME_FILEPATH_FOR_SED=${THEME_FILEPATH//\//\\/}

echo $SLACK_INTEROP_FILEPATH
echo $THEME_FILEPATH_FOR_SED

# Insert the CSS File Location in JS
sudo sed -i -e "s/$DARK_THEME_PATH_PLACEHOLDER/$THEME_FILEPATH_FOR_SED/g" $SLACK_INTEROP_FILEPATH

# Pack the Asar Archive for Slack
sudo npx asar pack $SLACK_RESOURCES_DIR/app.asar.unpacked $SLACK_RESOURCES_DIR/app.asar

cr_echo "Done! After executing this script, hit refresh (⌘ + R) or restart Slack for changes to take effect."
```

**Comment by patroza** [comment]
Nice. You have to fork the project here at github, and create the branch and push on your own repo, then you can open a pr to here

**Comment by LanikSJ** [comment]
I fixed most of those issues in the `slack-4.0` branch.  It's still a ⚠️ WIP ⚠️

**Comment by zoltaroth** [comment]
This is still broken for me as of this morning on 4.0.0 beta

I made sure the appended script is not part of the comment.  I added alerts into the function callback and those do get called but the theme does not go dark.

**Comment by LanikSJ** [comment]
I've seen some weirdness with refreshing the app so try restarting Slack. I tested it on my setup and it worked.

**Comment by zoltaroth** [comment]
> I've seen some weirdness with refreshing the app so try restarting Slack. I tested it on my setup and it worked.

Tried restarting.  Still no go.  Slack version is 4.0.0 beta.  I am going to try to do a clean install of slack and try again.  Any other debugging info that migt be helpful for you?

**Comment by dmzimmerman** [comment]
I just tried this on the 4.0.0 App Store version; it didn't work at all for me. It complains, when running the shell script, about not being able to find `/Applications/Slack.app/Contents/Resources/app.asar.unpacked/src/static/ssb-interop.js`.

**Comment by LanikSJ** [comment]
If you're getting that error you must not be on the `slack-4.0' branch.  You'll get that error only on `master'.

**Comment by dmzimmerman** [comment]
You're absolutely right; I could have sworn that I switched branches, but apparently I hadn't. Seems to be working now - thanks!

**Comment by LanikSJ** [comment]
No problem keep the bug reports coming.  I'm not going to merge #77 into the master until Slack 4.0 is released.

**Comment by zoltaroth** [comment]
Some success - after completely nuking all references to slack on my machine and doing a fresh install before starting slack for the first time this works.  However, after quitting and restarting slack it is back to the white bg mode again.  I am completely stumped.

**Comment by soleares** [comment]
I'm on the `slack-4.0` branch and the script ran without errors, but it doesn't change the style of the App Store 4.0.0 version.

**Comment by tolsadus** [comment]
`slack-4.0` branch, script runs fine, no errors or whatsoever (slack app store) but it doesn't change to a dark theme.

**Comment by zoltaroth** [comment]
narrowed down the restart issue to needing to delete: "$ rm -rf /Users/nandor/Library/Containers/com.tinyspeck.slackmacgap/Data/Library/Application\ Support/Slack/local-settings.json" in between restarts in order to get it to go dark again.

**Comment by tolsadus** [comment]
Works, but after a restart the bootSonic:"once" gets added back, and the dark theme goes away? (odd?)

**Comment by zoltaroth** [comment]
> Works, but after a restart the bootSonic:"once" gets added back, and the dark theme goes away? (odd?)

Yep you have to nuke it after each restart.

**Comment by tolsadus** [comment]
I've replaced the **once** with **never** and now it works! 🌵 Not sure if it's a reliable fix but well...

**Comment by zoltaroth** [comment]
> I've replaced the **once** with **never** and now it works! 🌵 Not sure if it's a reliable fix but well...

That seems to have worked for me as well.  I wish I knew what that setting was doing.
